python/dgllife/utils/splitters.py

```diff
@@ -484,23 +485,26 @@ def get_ordered_scaffold_sets(molecules, include_chirality, log_every_n):
         """
         if log_every_n is not None:
             print('Start computing Bemis-Murcko scaffolds.')
-        scaffolds = defaultdict(list)
+        scaffolds = {}
         for i, mol in enumerate(molecules):
             count_and_log('Computing Bemis-Murcko for compound',
                           i, len(molecules), log_every_n)
             # For mols that have not been sanitized, we need to compute their ring information
             try:
                 FastFindRings(mol)
-                mol_scaffold = MurckoScaffold.MurckoScaffoldSmiles(
-                    mol=mol, includeChirality=include_chirality)
+                mol_scaffold = AllChem.MurckoDecompose(mol)
                 # Group molecules that have the same scaffold
-                scaffolds[mol_scaffold].append(i)
+                if mol_scaffold not in scaffolds:
+                    scaffolds[mol_scaffold] = [i]
+                else:
+                    scaffolds[mol_scaffold].append(i)
```

Comment:
Any reasons for preferring this over `defaultdict(list)`?

Reply:
My bad, I added this for debugging. You can change it back to defaultdict